src/Model/Behavior/FootprintBehavior.php

```diff
@@ -163,9 +163,9 @@ protected function _injectEntity(EntityInterface $entity, ArrayObject $options,
         $new = $entity->isNew() !== false;
 
         foreach ($fields as $field => $when) {
-            if (!in_array($when, ['always', 'new', 'existing'])) {
+            if (!in_array($when, ['always', 'new', 'existing']) && !is_callable($when)) {
```

Comment:
Change all the `is_callable()` checks to `instanceof Closure` as it's much cheaper.